pkg/controller/noderefutil/providerid.go

```diff
@@ -39,13 +41,27 @@ func NewProviderID(id string) (*ProviderID, error) {
 		return nil, ErrEmptyProviderID
 	}
 
-	parsed, err := url.Parse(id)
-	if err != nil {
-		return nil, err
+	/*
+		- must start with at least one non-colon
+		- followed by ://
+		- followed by any number of characters
+		- must end with a non-slash
+	*/
+	re := regexp.MustCompile("^[^:]+://.*[^/]$")
```

Comment:
is there some regex we can import from k8s that would do this parsing for us so we don't have to maintain this regex?

Reply:
I did a quick search but I didn't find anything.

If we don't want a regex, we can loosen this a bit:
1. require the index of `://` be > 0
1. require the index of the last slash be after `://` and not the last character

Reply:
+1 on keeping the regex for validation purposes, the `regexp.MustCompile` can be as a global variable and only parsed once.

Reply:
Moved to a package var

Reply:
i prefer the regex as well, i was just hoping someone else had already defined it